onnxruntime/core/providers/cpu/math/transpose_matmul.cc

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "transpose_matmul.h"
+
+#include "core/framework/op_kernel_context_internal.h"
+#include "core/providers/cpu/math/matmul_helper.h"
+#include "core/util/math.h"
+
+namespace onnxruntime {
+namespace contrib {
+
+ONNX_OPERATOR_KERNEL_EX(
+    TransposeMatMul,
+    kMSDomain,
+    1,
+    kCpuExecutionProvider,
+    KernelDefBuilder().TypeConstraint("T", DataTypeImpl::GetTensorType<float>()),
+    TransposeMatMul);
+
+TransposeMatMul::TransposeMatMul(const OpKernelInfo& info)
+    : OpKernel{info} {
+  ORT_ENFORCE(info.GetAttr("transA", &trans_a_attr_).IsOK());
+  ORT_ENFORCE(info.GetAttr("transB", &trans_b_attr_).IsOK());
+}
+
+Status TransposeMatMul::Compute(OpKernelContext* context) const {
+  auto* context_internal = dynamic_cast<OpKernelContextInternal*>(context);
+  ORT_ENFORCE(context_internal, "Failed to get OpKernelContextInternal instance.");
```

Comment:
It's not necessary to use dynamic_cast as the operator threadpool moved to OpKernelContext so you shouldn't need to use OpKernelContextInternal at all.